src/architect/cluster/cluster.utils.ts

```diff
@@ -81,4 +87,23 @@ export default class ClusterUtils {
     }
     return cluster;
   }
+
+  private static async getClientVersion(kubeconfig_path: string): Promise<string> {
+    const { stdout } = await execa('kubectl', ['version', '--kubeconfig', kubeconfig_path, '--client', '--output', 'json']);
+    const { clientVersion } = JSON.parse(stdout);
+    return clientVersion.gitVersion;
+  }
+
+  public static async checkClientVersion(kubeconfig: string): Promise<void> {
+    const kubeconfig_path = untildify(kubeconfig);
+    const client_git_version = await this.getClientVersion(kubeconfig_path);
+    const client_semver = semver.coerce(client_git_version);
+    if (!client_semver) {
+      throw new ArchitectError(`Failed to translate Kubernetes cluster version ${client_git_version}.`);
```

Comment:
@mueschm should this be a hard error? If we get this wrong, the user can't create a cluster when it's possible that they should be able to. Should this be a yellow warning instead?

Reply:
If we can't verify the version we cannot guarantee it works. Better for them to contact support the set up a bad system. 1.21 is still prominent and showcased in a lot of online tutorials so I would prefer to be safe.